src/http/formdata/builder.cr

```diff
@@ -63,18 +69,21 @@ module HTTP::FormData
       @io << "\r\n\r\n"
       IO.copy(io, @io)
 
-      @state = :FIELD
+      @state = State::FIELD
     end
 
     # Finalizes the multipart message, this method must be called before the
     # generated multipart message written to the IO is considered valid.
     def finish : Nil
-      fail "Cannot finish form-data: no body parts" if @state == :START
-      fail "Cannot finish form-data: already finished" if @state == :FINISHED
-
-      @io << "\r\n--" << @boundary << "--"
-
-      @state = :FINISHED
+      case @state
+      when .start?
```

Comment:
Here we can use `in` and cover all cases

Reply:
Agreed. This is still unresolved.